### ISSUE TITLE
`[flag]` causes broken parsing when using `for` in `$if` scope.

### ISSUE THREAD

**Issue by Larpon** [vlang/v#13060]
The code below worked before 07cf6d9 (introduced with https://github.com/vlang/v/pull/12925)
```v
module main

[flag]
pub enum Fill {
    crash
}

fn font_path() string {
    $if macos {
        fonts := ['test','test/2']
        for font in fonts {
            if os.is_file(font) {
                return font
            }
        }
    }
    panic('failed to init font')
}


fn main() {
    println('I am here')
}
```
**V version:** 
OS: linux, Ubuntu 18.04.6 LTS
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i7-5600U CPU @ 2.60GHz
CC version: cc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0

getwd: ~/Projects/v
vmodules: ~/.vmodules
vroot: ~/Projects/v
vexe: ~/Projects/v/v
vexe mtime: 2022-01-06 12:46:28
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 fb66ec7

Git version: git version 2.17.1
Git vroot status: weekly.2021.52.1-67-gfb66ec7c (2 commit(s) behind V master)
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 3654d6a8

**What did you do?**
`v run code.v` (with contents above)

**What did you expect to see?**
`I am here`
 
**What did you see instead?**
```
V panic: sym: invalid type (typ=0 idx=0). Compiler bug. This should never happen. Please report the bug using `v bug file.v`.

v hash: cfb814a
/dev/shm/v_1000/v.758222383429161338.tmp.c:15895: at _v_panic: Backtrace
/dev/shm/v_1000/v.758222383429161338.tmp.c:37380: by v__ast__default_table_panic_handler
/dev/shm/v_1000/v.758222383429161338.tmp.c:37387: by v__ast__Table_panic
/dev/shm/v_1000/v.758222383429161338.tmp.c:38055: by v__ast__Table_sym
/dev/shm/v_1000/v.758222383429161338.tmp.c:61549: by v__gen__c__Gen_autofree_variable
/dev/shm/v_1000/v.758222383429161338.tmp.c:61535: by v__gen__c__Gen_autofree_scope_vars2
/dev/shm/v_1000/v.758222383429161338.tmp.c:61492: by v__gen__c__Gen_autofree_scope_vars_stop
/dev/shm/v_1000/v.758222383429161338.tmp.c:61477: by v__gen__c__Gen_autofree_scope_vars
/dev/shm/v_1000/v.758222383429161338.tmp.c:63632: by v__gen__c__Gen_return_stmt
/dev/shm/v_1000/v.758222383429161338.tmp.c:60524: by v__gen__c__Gen_stmt
/dev/shm/v_1000/v.758222383429161338.tmp.c:60192: by v__gen__c__Gen_stmts_with_tmp_var
/dev/shm/v_1000/v.758222383429161338.tmp.c:60140: by v__gen__c__Gen_stmts
/dev/shm/v_1000/v.758222383429161338.tmp.c:1944: by v__gen__c__Gen_gen_fn_decl
/dev/shm/v_1000/v.758222383429161338.tmp.c:1612: by v__gen__c__Gen_fn_decl
/dev/shm/v_1000/v.758222383429161338.tmp.c:60392: by v__gen__c__Gen_stmt
/dev/shm/v_1000/v.758222383429161338.tmp.c:60192: by v__gen__c__Gen_stmts_with_tmp_var
/dev/shm/v_1000/v.758222383429161338.tmp.c:60140: by v__gen__c__Gen_stmts
/dev/shm/v_1000/v.758222383429161338.tmp.c:59295: by v__gen__c__Gen_gen_file
/dev/shm/v_1000/v.758222383429161338.tmp.c:58936: by v__gen__c__gen
/dev/shm/v_1000/v.758222383429161338.tmp.c:19285: by v__builder__cbuilder__gen_c
/dev/shm/v_1000/v.758222383429161338.tmp.c:19295: by v__builder__cbuilder__build_c
/dev/shm/v_1000/v.758222383429161338.tmp.c:19267: by v__builder__cbuilder__compile_c
/dev/shm/v_1000/v.758222383429161338.tmp.c:18260: by v__builder__compile
/dev/shm/v_1000/v.758222383429161338.tmp.c:19423: by main__main
/dev/shm/v_1000/v.758222383429161338.tmp.c:20201: by main
```

**Comment by spytheman** [comment]
Should be fixed in 8088f46 .

**Comment by Larpon** [comment]
> Should be fixed in [8088f46](https://github.com/vlang/v/commit/8088f462c95b3facd6a089529d70a97b8570d312) .

:100:
